### PR TITLE
dev-libs/zziplib: 0.13.71_p20201021 rebase to master

### PR DESCRIPTION
**Pull request by vaukai**
Rebased from upstream develop to master
-MY_COMMIT="223930775aa5b325f04cec01f0b18726a7918821"
+MY_COMMIT="3921fc43bca7283f126bfb2e47ec7e7e24b5a5ea" # master Oct 21, 2020

Fixed manpages with single patches from upstream's develop branch
Removed 'test' USE flag since tests require internet access.
 ( gdraheim/zziplib#24 )

Package-Manager: Portage-3.0.4, Repoman-3.0.1
Signed-off-by: Volkmar W. Pogatzki <gentoo@pogatzki.net>

### CODE REVIEW

**Comment by vaukai** [comment]
Part of changes sent to upstream https://github.com/gdraheim/zziplib/pull/103

**Comment by vaukai** [comment]
Help wanted
Somehow it should be possible to convert the "[add_custom_target(check DEPENDS checks tests)](https://github.com/gentoo/gentoo/pull/17636/files#diff-0d4a365db40a67b8c3d16198cab8b402R63)" into an  "[add_test()](https://cmake.org/cmake/help/latest/command/add_test.html)" so that also the remaining "add_custom_target(checks" can be run via CTest.
Any help is highly appreciated

File: dev-libs/zziplib/zziplib-0.13.71_p20200419.ebuild [hunk 1]
```diff
@@ -35,26 +32,39 @@ RDEPEND="${DEPEND}"
 S="${WORKDIR}/${PN}-${MY_COMMIT}"
 
 PATCHES=(
-	"${FILESDIR}"/${PN}-0.13.69-009-perror.patch
-	"${FILESDIR}"/${PN}-0.13.71-installing-man3-pages.patch
+	"${FILESDIR}"/${PN}-0.13.69-009-perror.patch # not clear why this patch is needed
```
Comment (juippis) [inline dev-libs/zziplib/zziplib-0.13.71_p20200419.ebuild:35]: This is not really convincing, please study the patch and update why it's needed. Or remove this line/patch if it's not needed.

File: dev-libs/zziplib/Manifest [hunk 2]
```diff
@@ -1,2 +1,2 @@
 DIST zziplib-0.13.71.tar.gz 1150654 BLAKE2B 00394de5da8b3c5b95ef0aea060e523354845261b846e3c876b13d2d05c6569496475a279d6b211c3215a6f17e6bfe53958baabcebb13cd72d30a9f0c40d9449 SHA512 e035d0ac26dca78335ae3defc652543ff7b353a1a95d76ed1beeb21a08e16f287a62d488f528cfbb77d5b558581b68d439aa0823577524e9aa61a3cf5f208cb5
-DIST zziplib-0.13.71_p20200419.tar.gz 1153032 BLAKE2B 34aa38fde43805636a83f34c064ab09880b7200c414d3666786b7d750a26d8f39dbb431dde7cb81e785cd20a0d444e2a7308b5c79eb71bf0e9f52a051fe01883 SHA512 7f94524ddd745d81656ea9b0314db791e636ed9a8305c4fde7f88572b0302bff4c7f8fa76ccf368b7b8b1c8062a4ad85dfab8f30cdf6ee37f835ff5758751222
+DIST zziplib-0.13.71_p20200419.tar.gz 1151538 BLAKE2B 1c0493284e092bd98af9fd9f3e470d019d5b15ccbd110b5fa9d1cfe2e40fff29d000fa364f4f9539397d2fa196287703795823d36273479bc5c7a34be0a3a52d SHA512 61b12fa8e316c43aaa2a2a328e8df7c757dc5eda01fe629af57341f51e7f063d8f7d0be3bdd4c6f1667acdccc9a110a822e7687de1c6a388493bce828729f8d3
```
Comment (juippis) [inline dev-libs/zziplib/Manifest:2]: AFAIK the mirrors won't be updated unless you rename the tarball differently, just add something like `_upd.tar.gz` or so.

**Review by juippis — COMMENTED** [review]
```
ERROR: test_59742_zzipdir_mem_CVE_2017_5974 (__main__.ZZipTest)
run unzzip-mem -l $(CVE_2017_5974).zip
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/tmp/portage/gpm/portage/dev-libs/zziplib-0.13.71_p20200419/work/zziplib-c3de4fe291ab6756593a0ef6bad007ddd3953e00/test/zziptests.py", line 1703, in test_59742_zzipdir_mem_CVE_2017_5974
    run = shell("{exe} -l {tmpdir}/{filename} ".format(**locals()),
  File "/var/tmp/portage/gpm/portage/dev-libs/zziplib-0.13.71_p20200419/work/zziplib-c3de4fe291ab6756593a0ef6bad007ddd3953e00/test/zziptests.py", line 102, in shell
    raise subprocess.CalledProcessError(run.returncode, sh_command, output)
subprocess.CalledProcessError: Command '../bins/unzzip-mem -l tmp.test_59742/00150-zziplib-heapoverflow-__zzip_get32 ' returned non-zero exit status 2.

======================================================================
ERROR: test_59743_zzipdir_mix_CVE_2017_5974 (__main__.ZZipTest)
run unzzip-mix -l $(CVE_2017_5974).zip
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/tmp/portage/gpm/portage/dev-libs/zziplib-0.13.71_p20200419/work/zziplib-c3de4fe291ab6756593a0ef6bad007ddd3953e00/test/zziptests.py", line 1721, in test_59743_zzipdir_mix_CVE_2017_5974
    run = shell("{exe} -l {tmpdir}/{filename} ".format(**locals()),
  File "/var/tmp/portage/gpm/portage/dev-libs/zziplib-0.13.71_p20200419/work/zziplib-c3de4fe291ab6756593a0ef6bad007ddd3953e00/test/zziptests.py", line 102, in shell
    raise subprocess.CalledProcessError(run.returncode, sh_command, output)
subprocess.CalledProcessError: Command '../bins/unzzip-mix -l tmp.test_59743/00150-zziplib-heapoverflow-__zzip_get32 ' returned non-zero exit status 2.
...
...
...
======================================================================
FAIL: test_65474 (__main__.ZZipTest)
unzzip-zap -l $(CVE).zip
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/tmp/portage/gpm/portage/dev-libs/zziplib-0.13.71_p20200419/work/zziplib-c3de4fe291ab6756593a0ef6bad007ddd3953e00/test/zziptests.py", line 3263, in test_65474
    self.assertTrue(greps(run.errors, "Zipfile corrupted"))
AssertionError: [] is not true

======================================================================
FAIL: test_65479 (__main__.ZZipTest)
check $(CVE).zip
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/tmp/portage/gpm/portage/dev-libs/zziplib-0.13.71_p20200419/work/zziplib-c3de4fe291ab6756593a0ef6bad007ddd3953e00/test/zziptests.py", line 3276, in test_65479
    self.assertEqual(size, 82347)
AssertionError: 0 != 82347

----------------------------------------------------------------------
Ran 203 tests in 8.670s

FAILED (failures=42, errors=37, skipped=8)
ninja: build stopped: subcommand failed.
 * ERROR: dev-libs/zziplib-0.13.71_p20200419::testworld failed (test phase):
 *   ninja -v -j16 -l16 check failed
```

What's the problem and goal here? If you need to provide 10 patches to fix test case, that seems very wrong. Does the program work at all? Is upstream of no help?

File: dev-libs/zziplib/zziplib-0.13.71_p20200419.ebuild (hunk 1 above)
Comment (vaukai) [reply dev-libs/zziplib/zziplib-0.13.71_p20200419.ebuild]: removed

File: dev-libs/zziplib/Manifest (hunk 2 above)
Comment (vaukai) [reply dev-libs/zziplib/Manifest]: replaced with p20201021

**Comment by vaukai** [comment]
> ```
> ERROR: test_59742_zzipdir_mem_CVE_2017_5974 (__main__.ZZipTest)
> run unzzip-mem -l $(CVE_2017_5974).zip
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/var/tmp/portage/gpm/portage/dev-libs/zziplib-0.13.71_p20200419/work/zziplib-c3de4fe291ab6756593a0ef6bad007ddd3953e00/test/zziptests.py", line 1703, in test_59742_zzipdir_mem_CVE_2017_5974
>     run = shell("{exe} -l {tmpdir}/{filename} ".format(**locals()),
>   File "/var/tmp/portage/gpm/portage/dev-libs/zziplib-0.13.71_p20200419/work/zziplib-c3de4fe291ab6756593a0ef6bad007ddd3953e00/test/zziptests.py", line 102, in shell
>     raise subprocess.CalledProcessError(run.returncode, sh_command, output)
> subprocess.CalledProcessError: Command '../bins/unzzip-mem -l tmp.test_59742/00150-zziplib-heapoverflow-__zzip_get32 ' returned non-zero exit status 2.
> 
> ======================================================================
> ERROR: test_59743_zzipdir_mix_CVE_2017_5974 (__main__.ZZipTest)
> run unzzip-mix -l $(CVE_2017_5974).zip
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/var/tmp/portage/gpm/portage/dev-libs/zziplib-0.13.71_p20200419/work/zziplib-c3de4fe291ab6756593a0ef6bad007ddd3953e00/test/zziptests.py", line 1721, in test_59743_zzipdir_mix_CVE_2017_5974
>     run = shell("{exe} -l {tmpdir}/{filename} ".format(**locals()),
>   File "/var/tmp/portage/gpm/portage/dev-libs/zziplib-0.13.71_p20200419/work/zziplib-c3de4fe291ab6756593a0ef6bad007ddd3953e00/test/zziptests.py", line 102, in shell
>     raise subprocess.CalledProcessError(run.returncode, sh_command, output)
> subprocess.CalledProcessError: Command '../bins/unzzip-mix -l tmp.test_59743/00150-zziplib-heapoverflow-__zzip_get32 ' returned non-zero exit status 2.
> ...
> ...
> ...
> ======================================================================
> FAIL: test_65474 (__main__.ZZipTest)
> unzzip-zap -l $(CVE).zip
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/var/tmp/portage/gpm/portage/dev-libs/zziplib-0.13.71_p20200419/work/zziplib-c3de4fe291ab6756593a0ef6bad007ddd3953e00/test/zziptests.py", line 3263, in test_65474
>     self.assertTrue(greps(run.errors, "Zipfile corrupted"))
> AssertionError: [] is not true
> 
> ======================================================================
> FAIL: test_65479 (__main__.ZZipTest)
> check $(CVE).zip
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/var/tmp/portage/gpm/portage/dev-libs/zziplib-0.13.71_p20200419/work/zziplib-c3de4fe291ab6756593a0ef6bad007ddd3953e00/test/zziptests.py", line 3276, in test_65479
>     self.assertEqual(size, 82347)
> AssertionError: 0 != 82347
> 
> ----------------------------------------------------------------------
> Ran 203 tests in 8.670s
> 
> FAILED (failures=42, errors=37, skipped=8)
> ninja: build stopped: subcommand failed.
>  * ERROR: dev-libs/zziplib-0.13.71_p20200419::testworld failed (test phase):
>  *   ninja -v -j16 -l16 check failed
> ```
> 
> What's the problem and goal here? If you need to provide 10 patches to fix test case, that seems very wrong. Does the program work at all? Is upstream of no help?

Did not get such errors and cannot reproduce them:
----------------------------------------------------------------------
Ran 203 tests in 25.988s

OK (skipped=8)
>>> Completed testing dev-libs/zziplib-0.13.71_p20201021

**Comment by juippis** [comment]
Still getting the same error. Since they work for you, there must be some dependency that's missing. 
https://dev.gentoo.org/~juippis/tmp/zziplib-0.13.71_p20201021_build.log.txt full build.log.

**Comment by vaukai** [comment]
As I cannot reproduce your errors, could you please test with adding -j1 to "cmake_build check"?

**Comment by juippis** [comment]
Still fails. Reading the build.log, it says something like:
```
WARNING:test:ERR: unzip:  cannot find zipfile directory in one of tmp.test_59770/00153-zziplib-invalidread-zzip_mem_entry_extra_block or
WARNING:test:ERR:         tmp.test_59770/00153-zziplib-invalidread-zzip_mem_entry_extra_block.zip, and cannot find tmp.test_59770/00153-zziplib-invalidread-zzip_mem_entry_extra_block.ZIP, period.

EFINFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Fasarubbo%2Fpoc%2Fblob%2Fmaster/00153-zziplib-invalidread-zzip_mem_entry_extra_block
WARNING:test:*09: unzip -l tmp.test_59770/00153-zziplib-invalidread-zzip_mem_entry_extra_block 
WARNING:test:OUT: Archive:  tmp.test_59770/00153-zziplib-invalidread-zzip_mem_entry_extra_block
WARNING:test:ERR:   End-of-central-directory signature not found.  Either this file is not
WARNING:test:ERR:   a zipfile, or it constitutes one disk of a multi-part archive.  In the
WARNING:test:ERR:   latter case the central directory and zipfile comment will be found on
WARNING:test:ERR:   the last disk(s) of this archive.
WARNING:test:ERR: unzip:  cannot find zipfile directory in one of tmp.test_59770/00153-zziplib-invalidread-zzip_mem_entry_extra_block or
```

Soo is it trying to download something?

**Comment by vaukai** [comment]
> Soo is it trying to download something?

Just re-tested with network off: "FAILED (failures=42, errors=37, skipped=8)"
Could you give me a quick hint how to proceed?

**Comment by juippis** [comment]
Whatever it's trying to download, you may have to put in `SRC_URI=test? ( url )` for it to be available.

**Comment by vaukai** [comment]
> Whatever it's trying to download, you may have to put in `SRC_URI=test? ( url )` for it to be available.

Ah, thanks. 
Already trying to find out what it needs.  But it looks quite confusing to me
Guess I need to spend more time to have a closer look the next days

>>> Test phase: dev-libs/zziplib-0.13.71_p20201021
 * Working in BUILD_DIR: "/var/tmp/portage/dev-libs/zziplib-0.13.71_p20201021/work/zziplib-0.13.71_p20201021_build"
ninja -v -j8 -l0 check
[1/2] cd /var/tmp/portage/dev-libs/zziplib-0.13.71_p20201021/work/zziplib-0.13.71_p20201021_build/test && /bin/bash -c "../bins/zzcat test/README > test.out" && /bin/bash -c "if diff test.out /var/tmp/portage/dev-libs/zziplib-0.13.71_p20201021/work/zziplib-3921fc43bca7283f126bfb2e47ec7e7e24b5a5ea/README >/dev/null        ; then rm test.out ; echo check OK; true        ; else rm test.out ; echo check FAIL ; false; fi"
check OK
[2/2] cd /var/tmp/portage/dev-libs/zziplib-0.13.71_p20201021/work/zziplib-0.13.71_p20201021_build/test && /var/tmp/portage/dev-libs/zziplib-0.13.71_p20201021/temp/python3.9/bin/python /var/tmp/portage/dev-libs/zziplib-0.13.71_p20201021/work/zziplib-3921fc43bca7283f126bfb2e47ec7e7e24b5a5ea/test/zziptests.py --exeext= --topsrcdir=/var/tmp/portage/dev-libs/zziplib-0.13.71_p20201021/work/zziplib-3921fc43bca7283f126bfb2e47ec7e7e24b5a5ea -v
.............................................................................................INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Fasarubbo%2Fpoc%2Fblob%2Fmaster/00150-zziplib-heapoverflow-__zzip_get32
......INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Fasarubbo%2Fpoc%2Fblob%2Fmaster/00151-zziplib-heapoverflow-__zzip_get64
.....INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Fasarubbo%2Fpoc%2Fblob%2Fmaster/00152-zziplib-heapoverflow-zzip_mem_entry_extra_block
......INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Fasarubbo%2Fpoc%2Fblob%2Fmaster/00153-zziplib-invalidread-zzip_mem_entry_extra_block
......INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Fasarubbo%2Fpoc%2Fblob%2Fmaster/00156-zziplib-oobread-zzip_mem_entry_new
......INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Fasarubbo%2Fpoc%2Fblob%2Fmaster/00157-zziplib-nullptr-prescan_entry
......INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Fasarubbo%2Fpoc%2Fblob%2Fmaster/00154-zziplib-nullptr-zzip_mem_entry_new
.....INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Fasarubbo%2Fpoc%2Fblob%2Fmaster/00161-zziplib-assertionfailure-seeko_C
......INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2FProbeFuzzer%2Fpoc%2Fblob%2Fmaster%2Fzziplib/zziplib_0-13-67_zzdir_invalid-memory-access_main.zip
.......INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2FProbeFuzzer%2Fpoc%2Fblob%2Fmaster%2Fzziplib/zziplib_0-13-67_unzzip_infinite-loop_unzzip_cat_file.zip
......INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2FProbeFuzzer%2Fpoc%2Fblob%2Fmaster%2Fzziplib/zziplib_0-13-67_unzip-mem_buffer-access-with-incorrect-length-value_zzip_disk_fread.zip
......INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2FProbeFuzzer%2Fpoc%2Fblob%2Fmaster%2Fzziplib/zziplib_0-13-67_zzdir_memory-alignment-errors___zzip_fetch_disk_trailer.zip
.......INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2FProbeFuzzer%2Fpoc%2Fblob%2Fmaster%2Fzziplib/zziplib_0-13-67_unzip-mem_memory-alignment-errors_zzip_disk_findfirst.zip
......INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2FProbeFuzzer%2Fpoc%2Fblob%2Fmaster%2Fzziplib/zziplib_0-13-67_unzzip_memory-aligment-errors___zzip_fetch_disk_trailer.zip
......INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2FProbeFuzzer%2Fpoc%2Fblob%2Fmaster%2Fzziplib/zziplib_0-13-67_unzip-mem_memory-alignment-errors_zzip_disk_findfirst_64.zip
......INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Ffantasy7082%2Fimage_test%2Fblob%2Fmaster/c006-unknown-add-main
INFO:test:OUT  12336 zipped    61535% 0

INFO:test:ERR 
.INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Ffantasy7082%2Fimage_test%2Fblob%2Fmaster/c008-main-unknown-de
INFO:test:OUT  12336 defl:N    61535% 000000000
 12336 defl:N    61535% 00000000000000000000
245772K defl:N    -221% 0000000000
 2060K zipped    -38224% 00000000
 12336 stored    61535% 000000000000
 12336 stored    61535% 0000000000000000
 12336 defl:N    61535% 00000p000000000

INFO:test:ERR 
.INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Fret2libc%2F---provided-by-email---/poc_bypass_fix2.zip
ssssssINFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Ffantasy7082%2Fimage_test%2Fblob%2Fmaster/c005-bus-zzip_parse_root_directory
.INFO:test:OUT 
INFO:test:ERR did not open tmp.test_65461/c005-bus-zzip_parse_root_directory: tmp.test_65461/c005-bus-zzip_parse_root_directory: Invalid or incomplete multibyte or wide character

.INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Ffantasy7082%2Fimage_test%2Fblob%2Fmaster/003-unknow-def-zip
.....INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Ffantasy7082%2Fimage_test%2Fblob%2Fmaster/002-mem-leaks-zip
.INFO:test:need tmp.download/https%3A%2F%2Fgithub.com%2Fgdraheim%2Fzziplib%2Ffiles%2F2415382/evil.zip
.ERROR:test:FIXME: unzip-mem test_65485 is not solved
s.s
----------------------------------------------------------------------
Ran 203 tests in 26.623s

OK (skipped=8)
>>> Completed testing dev-libs/zziplib-0.13.71_p20201021

File: dev-libs/zziplib/zziplib-0.13.71_p20201021.ebuild [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	https://github.com/fantasy7082/image_test/blob/master/002-mem-leaks-zip
+	https://github.com/gdraheim/zziplib/files/2415382/evil.zip
+)"
+
+LICENSE="|| ( LGPL-2.1 MPL-1.1 )"
+SLOT="0/13"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
+IUSE="sdl static-libs test"
+
+# RESTRICT="!test? ( test )"
```
Comment (DarthGandalf) [inline dev-libs/zziplib/zziplib-0.13.71_p20201021.ebuild:43]: No need. It's already set to 'test' below.

File: dev-libs/zziplib/zziplib-0.13.71_p20201021.ebuild [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+# Tests require internet access
+# https://github.com/gdraheim/zziplib/issues/24
+RESTRICT="test"
```
Comment (thesamesam) [inline dev-libs/zziplib/zziplib-0.13.71_p20201021.ebuild:46]: If only we [had](https://bugs.gentoo.org/553696) EAPI 8 now!

**Comment by juippis** [comment]
As I said in my first comment, if fixing tests requires this much to do, it'd perhaps be best to restrict them. HOWEVER!

You've already come this far. Remember that the test files are now downloaded into `${DISTDIR}`, the build system probably can't find them there. Try copying the test zip files in `src_prepare` to a location where the test unit expects to find them.

**Comment by vaukai** [comment]
> [...] Try copying the test zip files in `src_prepare` to a location where the test unit expects to find them.

The location is "work/zziplib-0.13.71_p20201021_build/test/tmp.download/" and the downloaded stuff is automagically moved there during the the test phase.  Yes, at first I created the mkdir-and-copy block in 'src_prepare' and it still failed.  Taking deeper look into it I saw that the stuff was already there, without my mkdir-and-copy. 

I am going to remove all 'test' related stuff from the ebuild just leaving the comment '# https://github.com/gdraheim/zziplib/issues/24'.

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**